### PR TITLE
Gracefully handle LocaleProvider func returning null

### PR DESCRIPTION
**Pull request by samcook**
Gracefully handle the case where the LocaleProvider func returns null (defaults to en-US)

### CODE REVIEW

File: source/IdentityServer3.Localization/Internals/LocalizationServiceFactory.cs [hunk 1]
```diff
@@ -12,6 +12,11 @@ public static ILocalizationService Create(LocaleOptions options, IDictionary<str
         {
             var locale = options.GetLocale(env);
 
+            if (string.IsNullOrWhiteSpace(locale))
```
Comment (johnkors) [inline source/IdentityServer3.Localization/Internals/LocalizationServiceFactory.cs]: Absolutely, right. This was  a miss. Thanks! Could you move this logic into the `GetLocale` method?

**Comment by samcook** [comment]
Logic moved to `GetLocale` as requested.

File: source/IdentityServer3.Localization/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -33,3 +34,5 @@
 // [assembly: AssemblyVersion("1.0.*")]
 [assembly: AssemblyVersion("1.0.0.0")]
 [assembly: AssemblyFileVersion("1.0.0.0")]
+
+[assembly: InternalsVisibleTo("Unittests")]
```
Comment (johnkors) [inline source/IdentityServer3.Localization/Properties/AssemblyInfo.cs]: Sorry, I don't allow testing of internals. The unit tests are only allowed thru what's public. :/

File: source/Unittests/TheService.cs [hunk 3]
```diff
@@ -99,6 +99,23 @@ public void DoesNotThrowsExceptionForUnknownLocales(string locale)
             var dontCare = service.GetString(IdSrvConstants.Messages, MessageIds.MissingClientId);
         }
 
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void GetLocaleReturnsDefaultWhenLocaleProviderReturnsNullOrWhitespace(string locale)
```
Comment (johnkors) [inline source/Unittests/TheService.cs]: Could you change this to a unit test that uses the public method `GetString(string, string)` and verifies that a string is returned, even when the func is setup to returns nulls instead?

Comment (samcook) [reply source/Unittests/TheService.cs]: This was already covered by `DoesNotThrowsExceptionForUnknownLocales` (except for checking the returned string) - I renamed that test to `UnknownLocalesUseDefaultLocale` and check the returned result.

**Comment by johnkors** [comment]
Perfect! Thanks!

**Comment by johnkors** [comment]
Released as https://www.nuget.org/packages/IdentityServer3.Localization/0.3.187

**Pull request merged**